### PR TITLE
Switching attachment encryption to use an interface so we can configure the encryption

### PR DESCRIPTION
**Pull request by elevine**
This PR introduces the ISymmetricKey interface which referenced when encrypting and decrypting attachments. With this in place, we can then introduce a new implementation that uses AES in CTR mode so we can seek encrypted files. This is needed to support video playback of encrypted attachments

The existing tests continue to pass with these changes,

### CODE REVIEW

**Comment by elevine** [comment]
This has been pulled out of PR  https://github.com/APXLabs/couchbase-lite-net/pull/1.  Its the first part of what is needed.

File: src/Couchbase.Lite.Shared/Store/ISymmetricKey.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        /// </summary>
+        byte[] DecryptData(byte[] encryptedData);
+
+        /// <summary>
+        /// Streaming decryption.
+        /// </summary>
+        Stream DecryptStream(Stream stream);
+
+        /// <summary>
+        /// Creates a strem that will encrypt the given base stream
```
Comment (Ariemeth) [inline src/Couchbase.Lite.Shared/Store/ISymmetricKey.cs]: should be stream not strem

File: src/Couchbase.Lite.Shared/Store/ISymmetricKey.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        /// </summary>
+        /// <returns>The stream to write to for encryption</returns>
+        /// <param name="baseStream">The stream to read from</param>
```
Comment (Ariemeth) [inline src/Couchbase.Lite.Shared/Store/ISymmetricKey.cs]: Is the "from" in the comment necessary?

File: src/Couchbase.Lite.Shared/Store/SymmetricKey.cs [hunk 3]
```diff
@@ -251,7 +251,7 @@ public Stream DecryptStream(Stream stream)
         /// </summary>
         /// <returns>The stream to write to for encryption</returns>
         /// <param name="baseStream">The stream to read from</param>
-        public CryptoStream CreateStream(Stream baseStream)
+        public Stream CreateStream(Stream baseStream)
```
Comment (Ariemeth) [inline src/Couchbase.Lite.Shared/Store/SymmetricKey.cs:254]: Why the change from CryptoStream to Stream?

Comment (elevine) [inline src/Couchbase.Lite.Shared/Store/SymmetricKey.cs:254]: With our AES CTR implementation, we need to return something that isn't a CryptoStream.  Using the more generic Stream gives us that flexibility.

Comment (Ariemeth) [inline src/Couchbase.Lite.Shared/Store/SymmetricKey.cs:254]: Sounds good.

**Comment by orenfromberg** [comment]
:shipit:

File: src/Couchbase.Lite.Shared/Couchbase.Lite.Shared.projitems [hunk 4]
```diff
@@ -17,6 +17,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)Replication.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Replication\SSLGenerator.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Revision.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Store\ISymmetricKey.cs" />
```
Comment (davidlongest) [inline src/Couchbase.Lite.Shared/Couchbase.Lite.Shared.projitems]: Could you move this down to where the rest of the Store items are?

**Comment by davidlongest** [comment]
Made a minor comment. Still good to :shipit:

**Comment by elevine** [comment]
Related to https://github.com/couchbase/couchbase-lite-net/issues/621
